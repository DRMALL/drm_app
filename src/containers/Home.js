import React, { Component } from 'react'
import { View, Text, Image, Button, StatusBar } from 'react-native'
import { primaryColor, loginBackgroundColor } from '../common/constants'
import TabBarItem from '../components/units/TabBarItem'
import store from '../utils/store'
import HomeList from '../components/HomeList'
import { checkToken } from '../utils/handleToken'
import { getPort } from '../utils/fetchMethod'
import { getNews } from '../apis'
import { homeList } from '../utils/virtualData'

const homeIconSelected = require('../images/tabbar_icons/tabbar_home_selected.png')
    , homeIconNormal = require('../images/tabbar_icons/tabbar_home_normal.png')

export default class Home extends Component {
  static navigationOptions = (props) => ({
    tabBarIcon: ({ focused, tintColor }) => (
      <TabBarItem
        tintColor={tintColor}
        focused={focused}
        normalImage={homeIconNormal} 
        selectedImage={homeIconSelected} 
      />
    )
  });

  constructor(props) {
    super(props)
    this.state = {
      isMounted: false,
      newsListData: {},
    }
  }

  componentDidMount() {
    this.setState({isMounted: true})
    this.getNewsList()
  }

  componentWillUnmount(){
    this.setState({isMounted: false})
  }

  getNewsList() {
    checkToken('drmAppToken')
    .then(async token => {
      let res = await getPort(`${getNews}?token=${token}`)
      if(res.code == 200) {
        if(this.state.isMounted) {
          this.setState({
            newsListData: res.data,
          })
        }
      }
    })
  }

  render() {
    let { newsListData } = this.state
    return(
      <View style={{backgroundColor: loginBackgroundColor}}>
        <StatusBar backgroundColor={primaryColor} />
        <HomeList data={newsListData} {...this.props} />
      </View>
    )
  }
}

