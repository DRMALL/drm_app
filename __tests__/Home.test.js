import 'react-native'
import React from 'react'
import Home from '../src/containers/Home'
import renderer from 'react-test-renderer';

test('Home test', () => {
  const tree = renderer.create(
    <Home />
  ).toJSON()
  expect(tree).toMatchSnapshot()
})