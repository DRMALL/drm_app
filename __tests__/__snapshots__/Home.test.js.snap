// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home test 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#F7F7F7",
    }
  }
>
  <View>
    <RCTScrollView
      dataSource={
        ListViewDataSource {
          "items": 0,
        }
      }
      enableEmptySections={true}
      refreshControl={
        <RefreshControlMock
          colors={
            Array [
              "#ff0000",
              "#00ff00",
              "#0000ff",
            ]
          }
          onRefresh={[Function]}
          progressBackgroundColor="#fff"
          refreshing={false}
          title="下拉刷新"
          titleColor="rgba(0,0,0,.6)"
        />
      }
      renderRow={[Function]}
      renderScrollComponent={[Function]}
    >
      <RCTRefreshControl />
      <View />
    </RCTScrollView>
  </View>
</View>
`;
